llvm/tools/spirv-tools/CMakeLists.txt

```diff
@@ -5,8 +5,8 @@ if (NOT LLVM_INCLUDE_SPIRV_TOOLS_TESTS)
   return()
 endif ()
 
-if (NOT "SPIRV" IN_LIST LLVM_TARGETS_TO_BUILD)
-  message(FATAL_ERROR "Building SPIRV-Tools tests is unsupported without the SPIR-V target")
+if (NOT "SPIRV" IN_LIST LLVM_TARGETS_TO_BUILD AND NOT "mlir" IN_LIST LLVM_ENABLE_PROJECTS)
```

Comment:
This looks like a hack -- we shouldn't be predicating this on what other projects are enabled. IMO, this should be a separate flag that tells cmake that spirv tools are available.

Reply:
Ok, if I understand you correctly I think this `if` can be dropped completely. The CMake flag to enable spirv-tools is `LLVM_INCLUDE_SPIRV_TOOLS_TESTS` and it is tested at the beginning of this file.

Reply:
yes, I think spirv tools being enabled or not should be independent from their uses